lxd/api_project.go

```diff
@@ -876,6 +876,20 @@ func projectDelete(d *Daemon, r *http.Request) response.Response {
 			return fmt.Errorf("Only empty projects can be removed")
 		}
 
+		// Remove any automatically cached images.
+		filterForCached := true
+		cachedImages, err := cluster.GetImages(ctx, tx.Tx(), cluster.ImageFilter{Project: &project.Name, Cached: &filterForCached})
+		if err != nil {
+			return err
+		}
+
+		for _, image := range cachedImages {
+			err = tx.DeleteImage(ctx, image.ID)
```

Comment:
hrm, i still think this is going to end up leaving image files potentially orphaned on the disk.

If we take `imageDelete()` as an example of what has to be called today in order to clear out the cached images, we can see that it checks if the image is referenced by other projects, and if not, it then deletes the image file on each cluster member using `imageDeleteFromDisk`.

So if we aren't doing this when deleting the project then we will leaving image files on the disk without a reference in the DB.

Separately I've spotted an opportunity to remove some duplication by calling `imageDeleteFromDisk` from 

https://github.com/canonical/lxd/blob/33eeb706884cd868e84f6f4a29d1657b57a32902/lxd/images.go#L2707-L2719